components/eppp_link/eppp_eth.c

```diff
@@ -26,10 +33,55 @@ typedef struct header {
 static const char *TAG = "eppp_ethernet";
 static bool s_is_connected = false;
 static esp_eth_handle_t *s_eth_handles = NULL;
-static uint8_t s_out_buffer[ETH_MAX_PACKET_SIZE];
 static uint8_t s_their_mac[ETH_ADDR_LEN];
 static uint8_t s_our_mac[ETH_ADDR_LEN];
 
+#ifndef USE_ETHERNET_INIT_COMPONENT
+static esp_err_t simple_init_deinit(bool init, struct eppp_config_ethernet_s *config, esp_eth_handle_t *handle_array[])
```

Comment:
Is there any significant advantage of using this init-deinit instead of separate constructor-destructor?